src/host/readDataCooked.hpp

```diff
@@ -49,6 +49,38 @@ class COOKED_READ_DATA final : public ReadData
         DoneWithCarriageReturn,
     };
 
+    // A helper struct to ensure we keep track of _dirtyBeg while the
+    // underlying _buffer is being modified by COOKED_READ_DATA.
+    struct BufferState
+    {
+        const std::wstring& Get() const noexcept;
+        std::wstring Extract() noexcept

```

Comment:
does this signature offer the right move guarantees? you don't need to return an rvalue ref, do you?

Reply:
Due to RVO this should be more efficient I think. It's also probably a bit safer since it cannot result in stale pointers.